src/api/v1/lobby.rs

```diff
@@ -55,11 +55,14 @@ impl IntoResponse for TryContributeResponse {
 pub(crate) async fn try_contribute(
     session_id: SessionId,
     Extension(store): Extension<SharedState>,
+    Extension(storage): Extension<PersistentStorage>,
     Extension(transcript): Extension<SharedTranscript>,
 ) -> TryContributeResponse {
     let store_clone = store.clone();
     let app_state = &mut store.write().await;
 
+    let uid: String;
+
     // 1. Check if this is a valid session. If so, we log the ping time
     {
```

Comment:
Instead of having the uid not defined, you can do :


let uid = {
...
info.token.unique_identifier().to_owned()
};

Reply:
This would make our scopes really confusing. The purpose of the scope below is not to produce a `uid`, it's mostly to handle RwLock.

Reply:
Right, I am mostly not for declaring and not defining a variable, though it is a compile-time error most of the time: https://doc.rust-lang.org/rust-by-example/variable_bindings/declare.html#declare-first

Reply:
It's a compile-time error when you use before initialization, and it's a compile-time warning if you don't use at all => you need to initialize it properly anyway. :)

I agree that declaration without initialization is a bit harder to read but with the current way of locking it seemed like a good compromise. I'm open to other implementation ideas.

Reply:
Yep :) not blocking for this PR